src/pybind/mgr/prometheus/module.py

```diff
@@ -110,40 +110,30 @@ def health_status_to_number(status):
 class Metrics(object):
     def __init__(self):
         self.metrics = self._setup_static_metrics()
-        self.pending = {}
 
-    def set(self, key, value, labels=('',)):
+    def clear(self):
         '''
-        Set the value of a single Metrics. This should be used for static metrics,
-        e.g. cluster health.
+        Clear all the metrics data. This does not remove the initiated Metric classes.
         '''
-        self.metrics[key].set(value, labels)
+        for k in self.metrics.keys():
+            self.metrics[k].clear()
 
-    def append(self, key, value, labels = ('',)):
+    def set(self, key, value, labels=('',)):
         '''
-        Append a metrics to the staging area. Use this to aggregate daemon specific
-        metrics that can appear and go away as daemons are added or removed.
+        Set the value of a single Metric (with labels). Use this to set the value of any metric.
         '''
-        if key not in self.pending:
-            self.pending[key] = []
-        self.pending[key].append((labels, value))
+        self.metrics[key].set(value, labels)
 
-    def reset(self):
+    def all(self):
```

Comment:
This function call could be avoidedby simply returning the ```metrics``` object.

Reply:
I was thinking of removing the Metrics class altogether since now it is basically a simple dictionary. What about doing that?